pydata_sphinx_theme/_templates/switcher.html

```diff
@@ -0,0 +1,53 @@
+<div class="dropdown">
+    <button type="button" class="btn btn-primary btn-sm navbar-btn dropdown-toggle" id="switcher_button" data-toggle="dropdown">
+        {{ switcher_prefix }}{{ release }}
+        <span class="caret"></span>
+    </button>
+    <div id="switcher" class="dropdown-menu list-group-flush py-0" aria-labelledby="switcher_button">
+    <!-- dropdown will be populated by javascript on page load -->
+    </div>
+</div>
+
+<script type="text/javascript">
+// Function to construct the target URL from the JSON components
+function buildURL(template, entry) {
+    //
+    var parts = template.split(/[\{\}]/);
+    version_ix = parts.indexOf("version");
+    language_ix = parts.indexOf("language");
+    if (version_ix >= 0) {
+        parts[version_ix] = entry.version;
+    }
+    if (language_ix >= 0) {
+        parts[language_ix] = entry.language;
+    }
+    return parts.join("");
+}
+// Function to populate the (release, language) version switcher
+(function () {
+    // get JSON config
+    $.getJSON("{{ switcher_json_url }}", function(data, textStatus, jqXHR) {
+        // create the nodes first (before AJAX calls) to ensure the order is
+        // correct (for now, links will go to doc version homepage)
+        $.each(data, function(index, entry) {
+            entry.url = buildURL("{{ switcher_template_url }}", entry);
+            if (!("name" in entry)) {
+                entry.name = entry.version;
+            }
+            $("#switcher").append(`<a class="list-group-item list-group-item-action py-1" href="${entry.url}">${entry.name}</a>`);
+        });
+        // see if the current page exists in the other versions of the docs
+        $.each(data, function(index, entry) {
+            let filePath = "{{ pagename }}.html";
+            $.ajax({
+                type: 'HEAD',
+                url: `${entry.url}${filePath}`,
+                // update the target URL if it exists
+                success: function() {
+                    $("#switcher").children()[index].setAttribute("href", `${entry.url}${filePath}`);
+                }
+            });
```

Comment:
I think this implementation could be optimized a little. I understand that a HEAD request will be done per version each time a new page is opened, but maybe none of those links will be clicked. Imagine if there is a project with hundred active versions (yes! they exist 😸 )

So, I'd suggest always put the `href` but intercept the `onclick` event and do the check on the fly before redirecting.

Reply:
Besides, having a `fr` version shown in the switcher but not being able to click on it and not knowing why will be confusing, IMO.